### PR TITLE
chore(argo-workflows): removed deprecated Role rbac

### PR DESCRIPTION
**Pull request by danmx**
Signed-off-by: danmx <daniel@iziourov.info>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

**Review by jmeridth — APPROVED** [review]

**Comment by yu-croco** [comment]
v3.2.x is not supported anymore, so I think we can remove it :).
ref: https://argoproj.github.io/argo-workflows/releases/#supported-versions
image

File: charts/argo-workflows/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ appVersion: v3.4.4
 name: argo-workflows
 description: A Helm chart for Argo Workflows
 type: application
-version: 0.20.12
+version: 0.20.13
```
Comment (yu-croco) [inline charts/argo-workflows/Chart.yaml:6]: @danmx 
Though I approved this PR ,,,, this PR would affect some users, so maybe it's better to set `0.21.0` to let users know the impact? 🤔

Comment (danmx) [reply charts/argo-workflows/Chart.yaml:6]: ok, makes sense. I'll change it

**Review by yu-croco — APPROVED** [review]
Thank you!

**Pull request merged**